lib/src/main/java/io/ably/lib/realtime/Channel.java

```diff
@@ -383,8 +375,10 @@ private void onMessage(ProtocolMessage message) {
 			Message msg = messages[i];
 			try {
 				msg.decode(options);
+			} catch (MessageDecodeException e) {
+				Log.e(TAG, String.format("%s on channel %s", e.errorInfo.message, name));
 			} catch(AblyException e) {
-				Log.e(TAG, "Unexpected exception decrypting message", e);
+				Log.e(TAG, String.format("Unexpected exception decrypting message on channel %s", name), e);
```

Comment:
@paddybyers I saw the same thing, but then realised the previous code was like this before... I would agree we should instead catch `MessageDecodeException` if that is the exception we expect for this message, and perhaps another generic AblyException catch all?